x-pack/plugins/task_manager/server/queries/oldest_idle_action_task.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from 'kibana/server';
+
+/**
+ * Returns the timestamp of the oldest action task that may still be executed.
+ * Useful for cleaning up related objects that may no longer be needed.
+ * @internal
+ */
+export const getOldestIdleActionTask = async (
+  client: Pick<ElasticsearchClient, 'search'>,
+  taskManagerIndex: string
+): Promise<number | string> => {
+  let oldestAlertAction: number | string = Date.now();
```

Comment:
setting the default to `Date.now()` doesn't feel safe enough - if that value did end up getting used, inadvertently, it would delete all the action task objects, including ones queued to run.

Feels like `now-24h` would be a bit safer.  Would delete anything older than a day.  Which would delete queued action task objects that were somehow delayed from running for > 24hr, but if that's the case, there are bigger problems.

Reply:
Seems sensible to me 👍